resources/assets/js/components/ChartItemsList.vue

```diff
@@ -0,0 +1,86 @@
+        </div>
+        <div class="row list-group" id="chartItemAdder">
+            <div class="col">
+                <chart-item-edit v-for="(item, index) in currentStepData.labels" :key="index" :index-item="index"
+                                 :chart-item-label="item"
+                                 :chart-item-color="currentStepData.datasets[0].backgroundColor[index]"
+                                 :chart-item-value="currentStepData.datasets[0].data[index]"
+                                 :chart-item-reference="itemReferenceUpper[index]"
+                                 :available-results="availableResultsUpper"
+                                 @toggle1="selectCard($event)"
+                                 @refresh-chart-data-lower="refreshData($event)"
+                                 @remove-chart-item="toggleUpperRemoveChartItem($event)"></chart-item-edit>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import ChartItemEdit from "./ChartItemEdit.vue";
+
+export default {
+  components: {
+    ChartItemEdit
+  },
+  props: {
+    currentStepData: {},
+    itemReferenceUpper: {
+      type: Array
+    },
+    availableResultsUpper: {
+      type: Array
+    }
+  },
+  computed: {
+    addLabelButtonTitle() {
+      if (this.availableResultsUpper.length == 0)
+        return "No model-calculation results available to show. Please add a calculation in a parent-step.";
+      else return "Add a label to the graph to show the result of a calculation.";
+    }
+  },
+  methods: {
+    selectCard(index) {
+      for (let ind = 0; ind < this.currentStepData.labels.length; ind++) {
+        if (ind == index) $("#chartItemEditCollapse_" + ind).collapse("toggle");
+        else $("#chartItemEditCollapse_" + ind).collapse("hide");
+      }
+    },
+    addChartItem() {
+      this.currentStepData.labels.push("Enter Label");
+      this.currentStepData.datasets[0].backgroundColor.push("#00ff00");
+      this.currentStepData.datasets[0].data.push(17);
+      this.itemReferenceUpper.push({
+        reference: this.availableResultsUpper[0],
+        negation: false
+      });
+      this.$emit("refresh-chart-data1", this.currentStepData);
+      this.$emit("refresh-reference-data1", this.itemReferenceUpper);
+    },
+
+    refreshData(dataArray) {
+      let helpData = JSON.parse(JSON.stringify(dataArray));
+      this.currentStepData.labels[helpData[3]] = helpData[0];
+      this.currentStepData.datasets[0].backgroundColor[helpData[3]] = helpData[1];
+      this.currentStepData.datasets[0].data[helpData[3]] = Number(helpData[2]);
+      this.itemReferenceUpper[helpData[3]] = helpData[4];
+      this.$emit("refresh-chart-data", this.currentStepData);
+      this.$emit("refresh-reference-data", this.itemReferenceUpper);
+    },
+
+    toggleUpperRemoveChartItem(delIndex) {
+      let helpData = JSON.parse(JSON.stringify(this.currentStepData));
```

Comment:
Why?

Reply:
No real reason. While we were writing this code we had some trouble with pass-by-reference vs pass-by-value and forgot to remove the unnecessary hard-copies.